src/Process/LayeredMaterialNodeProcessing.js

```diff
@@ -168,24 +115,29 @@ export function updateLayeredMaterialNodeImagery(context, layer, node, parent) {
             // because even if this tile is outside of the layer, it could inherit it's
             // parent texture
             if (!layer.noTextureParentOutsideLimit &&
-                parent &&
                 parent.material &&
-                parent.material.indexOfColorLayer &&
-                parent.material.indexOfColorLayer(layer.id) >= 0) {
+                parent.material.getLayer &&
+                parent.material.getLayer(layer.id)) {
                 // ok, we're going to inherit our parent's texture
             } else {
                 node.layerUpdateState[layer.id].noMoreUpdatePossible();
                 return;
             }
         }
 
-        if (material.indexOfColorLayer(layer.id) === -1) {
-            material.pushLayer(layer, node.getCoordsForSource(layer.source));
-            const imageryLayers = context.view.getLayers(l => l.type === 'color');
-            const sequence = ImageryLayers.getColorLayersIdOrderedBySequence(imageryLayers);
+        if (!nodeLayer) {
+            // Create new MaterialLayer
+            const tileMT = layer.options.tileMatrixSet || extentsDestination[0].crs();
+            nodeLayer = material.addLayer(layer, tileMT);
+
+            // TODO: Sequence must be returned by parent geometry layer
```

Comment:
Why not do it now ?

Reply:
This PR is already big, it would take a new PR. I add a new issue.

Reply:
I open #922